Makefile

```diff
@@ -0,0 +1,50 @@
+PROJECT = cs_challenge
+
+# -------------------------------------------------------------------
+# Speedup fetching
+
+REBAR_GIT_CLONE_OPTIONS += --depth 1
+export REBAR_GIT_CLONE_OPTIONS
+
+# --------------------------------------------------------------------
+# Compilation.
+# --------------------------------------------------------------------
+
+# default mode
+ERLC_OPTS += +warn_export_all +warn_export_vars +warn_unused_import +warn_untyped_record
+ERLC_OPTS += +warn_missing_spec_all
+ERLC_OPTS += -Werror
+ERLC_OPTS += +debug_info
+
+# --------------------------------------------------------------------
+#  # Dependencies.
+# --------------------------------------------------------------------
+
+dep_thoas = git https://github.com/kawbo/thoas			feature/fix_typespec_of_json_term
+dep_cowboy = git https://github.com/ninenines/cowboy	2.12.0
+
+DEPS = thoas cowboy relx
+
+SHELL_DEPS = sync
+
+DEP_PLUGINS = cowboy relx
+
+PLT_APPS = ssl crypto stdlib inets public_key thoas ranch
+
+# --------------------------------------------------------------------
+#  # TDD Shell mode
+# --------------------------------------------------------------------
+
+SHELL_OPTS = -mode interactive -kernel shell_history enabled -pa ebin/ test/ -I -env ERL_LIBS deps -env LOGGER_CHARS_LIMIT unlimited -env LOGGER_DEPTH unlimited -eval 'sync:start(), sync:onsync(fun(Mods) -> [eunit:test(Mod) || Mod <- Mods] end), [lists:map(fun(Module) -> code:ensure_loaded(list_to_atom(lists:takewhile(fun(X) -> X /= 46 end, lists:subtract(Module,Dir)))) end, filelib:wildcard(Dir++"*.erl")) || Dir <- ["src/","test/"], ok =:= filelib:ensure_dir(Dir)]'
```

Comment:
a bit overkill :D

Reply:
😆 hah

I just hooking sync (https://github.com/rustyio/sync) into my dev environment via this way ^^

I simply want `make shell` and have ready-to-use TDD environment with automatic code recompilations, and automatic tests run when I'm changing codebase and automatic new modules loadings when I'm adding brand modules right during development.

There is new commit in sync enabling this functionality just as one function call in sync:
https://github.com/rustyio/sync/commit/964e071c4cf29ce53114d5def3d91aa0165c4937

but I didn't have chance to test it yet and I doubt it will glitch when new module introduced during dev session, but not loaded yet by the beam (you probably see that routine missing `code:ensure_loaded`)

anyway, that part of code is just my standard project setup